docs/design/host_to_services_OpenFlow.md

```diff
@@ -0,0 +1,215 @@
+
+To solve this problem, an additional load balancer is added with this implementation. The purpose of this new load balancer is to accommodate host endpoints. If endpoints are added for a service that contain a host endpoint, that VIP is moved to the new load balancer. Additionally, if one of those endpoints contain this node's IP address, it is replaced with the host's special masqueraded IP (IPv4: 169.254.169.2, IPv6: fd69::2).
+
+## Use Cases
+
+The following sections go over each potential traffic path originating from Host to Service. Note Host -> node port or external IP services are DNAT'ed in iptables to the cluster IP address before being sent out. Therefore the behavior of any host to service is essentially the same, forwarded towards the Cluster IP via the shared gateway bridge.
+
+For all of the following use cases, follow this topology:
+
+```text
+          host (ovn-worker, 172.18.0.3) 
+           |
+eth0----|breth0| ------ 172.18.0.3 OVN GR 100.64.0.4 --- join switch --- ovn_cluster_router --- 10.244.1.3 pod
+
+```
+
+The service used in the following use cases is:
+```text
+[trozet@trozet contrib]$ kubectl get svc web-service
+NAME          TYPE       CLUSTER-IP     EXTERNAL-IP   PORT(S)                       AGE
+web-service   NodePort   10.96.146.87   <none>        80:30015/TCP,9999:32326/UDP   4m54s
+
+```
+
+OVN masquerade IPs are 169.254.169.1, and fd69::1
+
+Another worker node exists, ovn-worker2 at 172.18.0.4.
+
+### OpenFlow Flows
+
+With the new implementation comes new OpenFlow rules in the shared gateway bridge. The following flows are added and used:
+```text
+ cookie=0xdeff105, duration=793.273s, table=0, n_packets=0, n_bytes=0, priority=500,ip,in_port="patch-breth0_ov",nw_src=172.18.0.3,nw_dst=169.254.169.2 actions=ct(commit,table=4,zone=64001,nat(dst=172.18.0.3))
+ cookie=0xdeff105, duration=793.273s, table=0, n_packets=0, n_bytes=0, priority=500,ip,in_port=LOCAL,nw_dst=169.254.169.1 actions=ct(table=5,zone=64002,nat)
+ cookie=0xdeff105, duration=793.273s, table=0, n_packets=0, n_bytes=0, priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.0/16 actions=ct(commit,table=2,zone=64001,nat(src=169.254.169.2))
+ cookie=0xdeff105, duration=793.273s, table=0, n_packets=0, n_bytes=0, priority=500,ip,in_port="patch-breth0_ov",nw_src=10.96.0.0/16,nw_dst=169.254.169.2 actions=ct(table=3,zone=64001,nat)
+
+ cookie=0xdeff105, duration=5.507s, table=2, n_packets=0, n_bytes=0, actions=mod_dl_dst:02:42:ac:12:00:03,output:"patch-breth0_ov"
```

Comment:
Why is `mod_dl_dst:02:42:ac:12:00:03` in both the flows for table 2 and 3? Aren't the flows in table 2 supposed to be converting the Destination MAC to the OVN-GWR's and the flows in table 3 converting the `mod_dl_dst` back to the host's mac? Or are they actually supposed to be the same?

Reply:
the macs of the GR and the host are the same